packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -220,6 +233,25 @@ function readPackageJson(packageName: string): PackageJson | undefined {
   }
 }
 
+function checkEmptyConfig(
+  userConfig: any,
+  {
+    showSolidityConfigWarnings = false,
+  }: { showSolidityConfigWarnings: boolean } = {
+    showSolidityConfigWarnings: false,
+  }
+) {
+  if (userConfig === undefined || Object.keys(userConfig).length === 0) {
+    let warning = `Hardhat config is returning an empty config object, check the export from the config file if this is unexpected.\n`;
+
+    if (!showSolidityConfigWarnings) {
+      warning += `Learn more about compiler configuration at https://hardhat.org/config"\n`;
```

Comment:
This one doesn't apply only to compilation.

```suggestion
      warning += `\nLearn more about configuring Hardhat at https://hardhat.org/config"\n`;
```